proxy/src/registry.rs

```diff
@@ -219,6 +233,29 @@ impl Registry {
             timestamp: SystemTime::now(),
         })
     }
+
+    /// Create a new unique user on the Registry.
+    pub async fn register_user(
+        &self,
+        _author: &ed25519::Pair,
+        handle: String,
+        id: String,
+    ) -> Result<Transaction, error::Error> {
+        let message_handle = registry::String32::from_string(handle)
+            .map_err(|_| error::UserValidation::HandleTooLong)?;
+        let message_id = registry::String32::from_string(id)
+            .map_err(|_| error::UserValidation::HandleTooLong)?;
+
+        Ok(Transaction {
+            id: registry::H256::random(),
```

Comment:
Strange to me that the id is an attribute of the transaction, and not just the content hash :eyes:

Reply:
We receive the id from calls to the registry client, we don' really want to know how it's computed, so we return it simple as a key to the caller.

Reply:
Ah, gotcha!